### PR TITLE
LUN-475: Long term exercise service

### PR DESCRIPTION
**Pull request by SteffiStuffel**
This pull request belongs to an issue on our [bugtracker](https://issues.tuerantuer.org/secure/RapidBoard.jspa?rapidView=27&view=detail).
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword _LUN_.

### CODE REVIEW

File: src/services/LongTermExerciseService.ts [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+  return saveWordNodeCards(wordNodeCards)
+}
+
+export const addWordToFirstSection = async (word: VocabularyItem): Promise<void> => addWordsToFirstSection([word])
+
+const moveWordToSection = async (wordNodeCard: WordNodeCard, targetSection: number): Promise<void> => {
+  const wordNodeCards: WordNodeCard[] = await getWordNodeCards()
+  const wordNodeCardToMove = wordNodeCards.find(item => JSON.stringify(item) === JSON.stringify(wordNodeCard))
+  if (wordNodeCardToMove) {
+    wordNodeCardToMove.section = sections[Math.min(Math.max(0, targetSection), sections.length - 1)]
```
Comment (f1sh1918) [inline src/services/LongTermExerciseService.ts:107]: can you explain why this is needed to get the correct section. I didn't get it yet tbh :/
i would expect `wordNodeCardToMove.section = sections[targetSection]` should be fine

**Review by f1sh1918 — APPROVED** [review]
Looks fine as far as i understood correct :)
I think since this is a kind of complex service i would appreciate some short comments to the functions

**Review by LeandraH — APPROVED** [review]
Nice!

File: src/services/LongTermExerciseService.ts [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+    wordNodeCardToMove.inThisSectionSince = new Date()
+    await saveWordNodeCards(wordNodeCards)
+  }
+}
+
+export const moveWordToNextSection = async (wordNodeCard: WordNodeCard): Promise<void> =>
+  moveWordToSection(wordNodeCard, wordNodeCard.section + 1)
+
+export const moveWordToPreviousSection = async (wordNodeCard: WordNodeCard): Promise<void> =>
+  moveWordToSection(wordNodeCard, wordNodeCard.section - 1)
```
[the remark this answers is not in the document]
Comment (SteffiStuffel) [reply src/services/LongTermExerciseService.ts:117]: There are different types of ways to handle incorrect words, more frequently used one, and the one we decided to use, is just to move it back one section. Your suggestion would also be find, but we decided to do it not like this. If you have a strong opinion on this, we can discuss about it. If so please open a discussion on mattermost, so Julia and Roman can also participate.

**Comment by SteffiStuffel** [comment]
> Looks fine as far as i understood correct :) I think since this is a kind of complex service i would appreciate some short comments to the functions

Just add comments in general is not a good practice in terms of clean code principles. If you think some functions are especially unclear, i can think of better naming.

File: src/services/LongTermExerciseService.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (LeandraH) [reply src/services/LongTermExerciseService.ts:119]: Cool, thanks for the explanation, works for me :)

**Pull request merged**